File: app/src/main/java/su/linka/pictures/Card.java
package su.linka.pictures;

import org.json.JSONException;
import org.json.JSONObject;

import java.util.Objects;

public class Card {
    final int id ;
    public String imagePath ;
    public String title ;
    public String audioPath ;
    public int cardType ;

    public Card(int id, int cardType){
        this.id = id;
        this.cardType = cardType;
    }
    public Card(int id, String imagePath, String title, String audioPath, int cardType) {
        this.id = id;
        this.imagePath = imagePath;
        this.title = title;
        this.audioPath = audioPath;
        this.cardType = cardType;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        Card card = (Card) o;
        return id == card.id && cardType == card.cardType && Objects.equals(imagePath, card.imagePath) && Objects.equals(title, card.title) && Objects.equals(audioPath, card.audioPath);
    }

    @Override
    public int hashCode() {
        return Objects.hash(id, imagePath, title, audioPath, cardType);
    }

    public JSONObject toJSONObject() throws JSONException {
        JSONObject card = new JSONObject();
        card.put("id", id)
                .put("title", title)
                .put("imagePath", imagePath)
                .put("audioPath", audioPath)
                .put("cardType", cardType);
        return card;
    }

    @Override
    public Card clone() throws CloneNotSupportedException {
        return (Card) super.clone();
    }
}

File: shared/src/androidMain/kotlin/su/linka/pictures/Card.kt
package su.linka.pictures

import org.json.JSONException
import org.json.JSONObject

data class Card(
    val id: Int,
    var imagePath: String? = null,
    var title: String? = null,
    var audioPath: String? = null,
    var cardType: Int = 0
) : Cloneable {

    constructor(id: Int, cardType: Int) : this(id, null, null, null, cardType)

    @Throws(JSONException::class)
    fun toJSONObject(): JSONObject {
        return JSONObject()
            .put("id", id)
            .put("title", title)
            .put("imagePath", imagePath)
            .put("audioPath", audioPath)
            .put("cardType", if (cardType < 3) cardType else 2)
    }

    public override fun clone(): Card = copy()
}
